src/hibernate-resume/hibernate-resume.c

```diff
@@ -94,6 +106,29 @@ static int setup_hibernate_info_and_warn(void) {
         return 1;
 }
 
+static int action_clear_efi(void) {
+        int r;
+
+        assert(arg_clear_efi);
+
+        if (in_initrd()) {
+                log_info("Running in initrd, --clear-efi has no effect, exiting.");
+                return 0;
+        }
```

Comment:
hmm, if people call this from the cmdline i think we should allow this to pass.

Reply:
i mean, key really is that we do not call it from the initrd *automatically*, but calling it explicitly should be fine.

hence this condition i think belongs in the unit file, not in the C code.

(we had similar cases in the past, where we ended up losening the C conditioning in favour of unit file based conditioning)